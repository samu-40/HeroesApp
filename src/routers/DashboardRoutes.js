import React from 'react'
import { Redirect, Route, Switch } from 'react-router'
import ScreenDC from '../components/dc/ScreenDC'
import HeroScreen from '../components/heroes/HeroScreen'
import ScreenMarvel from '../components/marvel/ScreenMarvel'
import SearchScreen from '../components/search/SearchScreen'
import { Navbar } from '../components/UI/Navbar'

const DashboardRoutes = () => {
    return (
        <>
            <Navbar/>

            <div className='super-div'>
                <Switch>
                    <Route exact path='/marvel' component={ScreenMarvel} />
                    <Route exact path='/hero/:heroeId' component={HeroScreen} />
                    <Route exact path='/dc' component={ScreenDC} />
                    <Route exact path='/search' component={SearchScreen} />

                    <Redirect to='/marvel' />
                </Switch>
            </div>
        </>
    )
}

export default DashboardRoutes
