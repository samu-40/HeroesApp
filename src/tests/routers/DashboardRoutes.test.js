import '../../setupTest';
import { mount } from "enzyme";
import DashboardRoutes from '../../routers/DashboardRoutes';
import { AuthContext } from '../../auth/AuthContext';
import { MemoryRouter } from 'react-router-dom';


describe('Pruebas en DashboardRoutes.js', () => {

    test('Debe mostarse correctamente ', () => {

        const contextValue = {
            dispatch: jest.fn(),
            user: { logged: false, name: 'Samuel' }
        };

        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <MemoryRouter>
                    <DashboardRoutes/>
                </MemoryRouter>
            </AuthContext.Provider>
        );

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('span').text().trim()).toBe('Samuel');

    });
    
});
