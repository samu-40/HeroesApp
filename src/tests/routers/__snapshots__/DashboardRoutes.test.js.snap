// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en DashboardRoutes.js Debe mostarse correctamente  1`] = `
Array [
  <nav
    className="navbar"
  >
    <a
      className="brand"
      href="/"
      onClick={[Function]}
    >
      Asociaciones
    </a>
    <div
      className="links"
    >
      <a
        aria-current="page"
        className="link active"
        href="/marvel"
        onClick={[Function]}
        style={Object {}}
      >
        Marvel
      </a>
      <a
        aria-current={null}
        className="link"
        href="/dc"
        onClick={[Function]}
      >
        DC
      </a>
      <a
        aria-current={null}
        className="link"
        href="/search"
        onClick={[Function]}
      >
        Search
      </a>
    </div>
    <div>
      <span>
        Samuel
      </span>
      <button
        onClick={[Function]}
      >
        Logout
      </button>
    </div>
  </nav>,
  <div
    className="super-div"
  >
    <div>
      <h1>
        Marvel Screen
      </h1>
      <div
        className="container-flex animate__animated animate__fadeIn"
      >
        <div
          className="main-container-flex"
        >
          <div
            className="img"
          >
            <img
              alt="Spider Man"
              className="picture"
              src="./assets/heroes/marvel-spider.jpg"
            />
          </div>
          <div
            className="text"
          >
            <h3>
              Spider Man
            </h3>
            <p>
              Peter Parker
            </p>
            <a
              href="/hero/marvel-spider"
              onClick={[Function]}
            >
              More...
            </a>
          </div>
        </div>
        <div
          className="main-container-flex"
        >
          <div
            className="img"
          >
            <img
              alt="Captain America"
              className="picture"
              src="./assets/heroes/marvel-captain.jpg"
            />
          </div>
          <div
            className="text"
          >
            <h3>
              Captain America
            </h3>
            <p>
              Steve Rogers
            </p>
            <a
              href="/hero/marvel-captain"
              onClick={[Function]}
            >
              More...
            </a>
          </div>
        </div>
        <div
          className="main-container-flex"
        >
          <div
            className="img"
          >
            <img
              alt="Iron Man"
              className="picture"
              src="./assets/heroes/marvel-iron.jpg"
            />
          </div>
          <div
            className="text"
          >
            <h3>
              Iron Man
            </h3>
            <p>
              Tony Stark
            </p>
            <a
              href="/hero/marvel-iron"
              onClick={[Function]}
            >
              More...
            </a>
          </div>
        </div>
        <div
          className="main-container-flex"
        >
          <div
            className="img"
          >
            <img
              alt="Thor"
              className="picture"
              src="./assets/heroes/marvel-thor.jpg"
            />
          </div>
          <div
            className="text"
          >
            <h3>
              Thor
            </h3>
            <p>
              Thor Odinson
            </p>
            <a
              href="/hero/marvel-thor"
              onClick={[Function]}
            >
              More...
            </a>
          </div>
        </div>
        <div
          className="main-container-flex"
        >
          <div
            className="img"
          >
            <img
              alt="Hulk"
              className="picture"
              src="./assets/heroes/marvel-hulk.jpg"
            />
          </div>
          <div
            className="text"
          >
            <h3>
              Hulk
            </h3>
            <p>
              Bruce Banner
            </p>
            <a
              href="/hero/marvel-hulk"
              onClick={[Function]}
            >
              More...
            </a>
          </div>
        </div>
        <div
          className="main-container-flex"
        >
          <div
            className="img"
          >
            <img
              alt="Wolverine"
              className="picture"
              src="./assets/heroes/marvel-wolverine.jpg"
            />
          </div>
          <div
            className="text"
          >
            <h3>
              Wolverine
            </h3>
            <p>
              James Howlett
            </p>
            <a
              href="/hero/marvel-wolverine"
              onClick={[Function]}
            >
              More...
            </a>
          </div>
        </div>
        <div
          className="main-container-flex"
        >
          <div
            className="img"
          >
            <img
              alt="Daredevil"
              className="picture"
              src="./assets/heroes/marvel-daredevil.jpg"
            />
          </div>
          <div
            className="text"
          >
            <h3>
              Daredevil
            </h3>
            <p>
              Matthew Michael Murdock
            </p>
            <a
              href="/hero/marvel-daredevil"
              onClick={[Function]}
            >
              More...
            </a>
          </div>
        </div>
        <div
          className="main-container-flex"
        >
          <div
            className="img"
          >
            <img
              alt="Hawkeye"
              className="picture"
              src="./assets/heroes/marvel-hawkeye.jpg"
            />
          </div>
          <div
            className="text"
          >
            <h3>
              Hawkeye
            </h3>
            <p>
              Clinton Francis Barton
            </p>
            <a
              href="/hero/marvel-hawkeye"
              onClick={[Function]}
            >
              More...
            </a>
          </div>
        </div>
        <div
          className="main-container-flex"
        >
          <div
            className="img"
          >
            <img
              alt="Cyclops"
              className="picture"
              src="./assets/heroes/marvel-cyclops.jpg"
            />
          </div>
          <div
            className="text"
          >
            <h3>
              Cyclops
            </h3>
            <p>
              Scott Summers
            </p>
            <a
              href="/hero/marvel-cyclops"
              onClick={[Function]}
            >
              More...
            </a>
          </div>
        </div>
        <div
          className="main-container-flex"
        >
          <div
            className="img"
          >
            <img
              alt="Silver Surfer"
              className="picture"
              src="./assets/heroes/marvel-silver.jpg"
            />
          </div>
          <div
            className="text"
          >
            <h3>
              Silver Surfer
            </h3>
            <p>
              Norrin Radd
            </p>
            <a
              href="/hero/marvel-silver"
              onClick={[Function]}
            >
              More...
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
